packages/cli/src/databases/db-connection.ts

```diff
@@ -80,15 +82,39 @@ export class DbConnection {
 		}
 	}
 
-	/** Ping DB connection every 2 seconds */
+	/** Ping DB connection every 10 seconds */
 	private scheduleNextPing() {
-		this.pingTimer = setTimeout(async () => await this.ping(), 2000);
+		this.pingTimer = setTimeout(async () => await this.ping(), 10 * Time.seconds.toMilliseconds);
 	}
 
 	private async ping() {
 		if (!this.dataSource.isInitialized) return;
+
 		try {
-			await this.dataSource.query('SELECT 1');
+			let dataSource = this.dataSource;
+
+			// Prevent re-using pool connections that might be stale (e.g. when db password changes)
+			if (this.options.type !== 'sqlite') {
+				const options = {
+					...this.options,
+					// Add these to minimize overhead
+					synchronize: false,
+					migrationsRun: false,
+					logging: false,
+				};
+				options.poolSize = 1;
```

Comment:
```suggestion
				const options = {
					...this.options,
					// Add these to minimize overhead
					synchronize: false,
					migrationsRun: false,
					logging: false,
					poolSize: 1,
				};
```

or maybe either use `pick` to pick only a smallest required subset of the options needed, or be even leaner and (if possible) create an instance of the `Driver` directly to avoid all the overhead of creating a new db connection in typeorm.